### ISSUE TITLE
Incorrect GPU memory usage 

### ISSUE THREAD

**Issue by a7b23** [tensorflow/tensorflow#14950]
with tf.device('/gpu:0') :
    	with tf.variable_scope('network1') :
    		x = tf.get_variable('inp1',shape = [256,50,50,3],dtype = tf.float32)
    		w1 = tf.get_variable('weight1',shape = [1,1,3,256],dtype = tf.float32)
    		y1 = tf.nn.conv2d(x, w1, strides=[1, 1, 1, 1], padding='VALID')
    		w1 = tf.get_variable('weight2',shape = [1,1,256,256],dtype = tf.float32)
    		y1 = tf.nn.conv2d(y1, w1, strides=[1, 1, 1, 1], padding='VALID')
    		w1 = tf.get_variable('weight3',shape = [1,1,256,1],dtype = tf.float32)
    		y1 = tf.nn.conv2d(y1, w1, strides=[1, 1, 1, 1], padding='VALID')
    	
    
    with tf.device('/gpu:0') :
    	with tf.variable_scope('network2') :
    		x1 = tf.get_variable('inp2',shape = [1,3],dtype = tf.float32)
    		w2 = tf.get_variable('weight4',shape = [3,8192],dtype = tf.float32)
    		y2 = tf.matmul(x1,w2)
    		w2 = tf.get_variable('weight5',shape = [8192,8192],dtype = tf.float32)
    		y2 = tf.matmul(y2,w2)
    		w2 = tf.get_variable('weight6',shape = [8192,1],dtype = tf.float32)
    		y2 = tf.matmul(y2,w2)

If I have the above model graph, then running it on a GPU and checking memory usage via nvidia-smi shows usage of 4200MB . However when running only the network1 and commenting out network2, nvidia-smi shows usage of 2200 MB. With only network2, it shows usage of 700MB.I have already set gpu_options.allow_growth = True

How is then the total graph using 4200 MB when it should have used only 2900(2200 + 700)MB ?

I am using Ubuntu 14.04, python version - 2.7 , tensorflow version - 1.2

**Comment by aselle** [comment]
You are probably not accounting for intermediates implied by the various operations? You can't just assume memory usage is going to be the total memory usage of the variables.

**Comment by a7b23** [comment]
I don't have any other operation apart from the ones in the graph. So when only network1 shows GPU usage of 2200 MB it does include all the intermediates , similarly with network2.

**Comment by a7b23** [comment]
I have another example showing the above concern

<img width="1205" alt="screen shot 2017-11-30 at 1 10 37 am" src="https://user-images.githubusercontent.com/12389081/33395462-96c2a630-d56b-11e7-83a3-d2670caa90b2.png">

In the graph defined in the image if I run it on a single GPU , the node network1 shows usage of 1.2 GB whereas if I run the left and right half of graphs on seperate GPU's the node network1 shows usage of 2 GB.

**Comment by aselle** [comment]
Well it could be that even though you have allow_growth on it, the allocator graphs a larger chunk that it needs. This is often done by memory allocators for efficiency and to reduce fragmentation. @zheng-xq, might be able to provide more insight.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 161 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 177 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 192 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 207 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 222 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 237 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 252 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 267 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 282 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 297 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 312 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 327 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 342 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 357 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by jvishnuvardhan** [comment]
This is a stale issue. Please check the issue with latest TensorFlow. If the issue still persists in the newer version of TF, please feel free to reopen it by providing details about the issue and a standalone code to reproduce the issue. Thanks!
